test/tools/cli/IonJavaCli.java

```diff
@@ -0,0 +1,1341 @@
+import tools.errorReport.ErrorDescription;
+import tools.errorReport.ErrorType;
+import tools.events.Event;
+import tools.events.EventType;
+import tools.events.ImportDescriptor;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IonJavaCli {
+    private static final int CONSOLE_WIDTH = 120; // Only used for formatting the USAGE message
+    private static final int USAGE_ERROR_EXIT_CODE = 1;
+    private static final int IO_ERROR_EXIT_CODE = 2;
+    private static final int BUFFER_SIZE = 128 * 1024;
+    private static final String SYSTEM_OUT_DEFAULT_VALUE = "out";
+    private static final String SYSTEM_ERR_DEFAULT_VALUE = "err";
+    private static final IonTextWriterBuilder ION_TEXT_WRITER_BUILDER = IonTextWriterBuilder.standard();
+    private static final IonSystem ION_SYSTEM = IonSystemBuilder.standard().build();
+    private static final String EMBEDDED_STREAM_ANNOTATION = "embedded_documents";
+    private static final String EVENT_STREAM = "$ion_event_stream";
+
+    public static void main(final String[] args) {
+        CommandArgs parsedArgs = new CommandArgs();
+        CmdLineParser parser = new CmdLineParser(parsedArgs);
+        parser.getProperties().withUsageWidth(CONSOLE_WIDTH);
+        ComparisonType comparisonType = null;
+        OutputFormat outputFormat = null;
+        CommandType commandType = null;
+
+        try {
+            //parse
+            parser.parseArgument(args);
+            validateArgs(parsedArgs);
+            //get value
+            comparisonType = parsedArgs.getComparisonType();
+            outputFormat = parsedArgs.getOutputFormat();
+            commandType = parsedArgs.getCommand();
+        } catch (CmdLineException | IllegalArgumentException e) {
+            printHelpText(e.getMessage(), parser, parsedArgs);
+            System.exit(USAGE_ERROR_EXIT_CODE);
+        }
+
+        if (commandType == CommandType.COMPARE) {
+            try (
+                    //Initialize output stream, never return null. (default value: STDOUT)
+                    OutputStream outputStream = initOutputStream(parsedArgs, SYSTEM_OUT_DEFAULT_VALUE, null);
+                    //Initialize error report, never return null. (default value: STDERR)
+                    OutputStream errorReportOutputStream =
+                            initOutputStream(parsedArgs, SYSTEM_ERR_DEFAULT_VALUE, null);
+                    IonWriter ionWriterForOutput = outputFormat.createIonWriter(outputStream);
+                    IonWriter ionWriterForErrorReport = outputFormat.createIonWriter(errorReportOutputStream);
+            ) {
+                compareFiles(ionWriterForOutput, ionWriterForErrorReport, parsedArgs, comparisonType);
```

Comment:
I would move the `commandType` check inside this `try` so you can reuse a single `try` block. No need to duplicate it in a branch for the other command type.